src/inverted_index.c

```diff
@@ -971,7 +974,13 @@ static bool IndexReader_ReadWithSeeker(IndexReader *ir, t_docId docId) {
   bool found = false;
   while (!found) {
     // try and find docId using seeker
-    found = ir->decoders.seeker(&ir->br, &ir->decoderCtx, ir, docId, ir->record);
+    IndexBlockReader reader = (IndexBlockReader){
+      .buffReader = ir->br,
+      .curBaseId = (ir->decoders.decoder != readRawDocIdsOnly) ? ir->lastId : IR_CURRENT_BLOCK(ir).firstId,
+    };
+    found = ir->decoders.seeker(&reader, &ir->decoderCtx, docId, ir->record);
+    ir->br = reader.buffReader;
```

Comment:
just for my education, this is because the reader may have "moved" to another block right?

Reply:
Wouldn't perhaps make more sense for the `IndexReader` to actually have an IndexBlockReader instead of  a BufferReader as member?